lib/typetraits.h

```diff
@@ -42,5 +42,7 @@
 	(T) ~type_max(T)                                                      \
 )
 
+#define is_string_literal(S)  (!!("" S ""))
```

Comment:
Hmm, actually, this should be called must_be_string_literal(), since it doesn't allow passing something else.

Reply:
And, since I actually want this for strdupa(3), I guess it's better to just define a STRDUPA() wrapper with this integrated.